source/Octopus.Tentacle.Tests.Integration/ClientScriptExecutionScriptServiceV2IsNotRetried.cs

```diff
@@ -0,0 +1,231 @@
+
+namespace Octopus.Tentacle.Tests.Integration
+{
+    [IntegrationTestTimeout]
+    public class ClientScriptExecutionScriptServiceV2IsNotRetried : IntegrationTest
+    {
+        [Test]
+        [TestCaseSource(typeof(TentacleTypesToTest))]
+        public async Task WhenANetworkFailureOccurs_DuringStartScript_WithRetriesDisabled_TheCallIsNotRetried(TentacleType tentacleType)
+        {
+            using var clientTentacle = await new ClientAndTentacleBuilder(tentacleType)
+                .WithRetriesDisabled()
+                .WithPortForwarderDataLogging()
+                .WithResponseMessageTcpKiller(out var responseMessageTcpKiller)
+                .WithRetryDuration(TimeSpan.FromMinutes(4))
+                .WithTentacleServiceDecorator(new TentacleServiceDecoratorBuilder()
+                    .LogCallsToScriptServiceV2()
+                    .RecordExceptionThrownInScriptServiceV2(out var scriptServiceExceptions)
+                    .CountCallsToScriptServiceV2(out var scriptServiceCallCounts)
+                    .DecorateScriptServiceV2With(new ScriptServiceV2DecoratorBuilder()
+                        .BeforeStartScript(() =>
+                        {
+                            if (scriptServiceExceptions.StartScriptLatestException == null)
+                            {
+                                responseMessageTcpKiller.KillConnectionOnNextResponse();
+                            }
+                        })
+                        .Build())
+                    .Build())
+                .Build(CancellationToken);
+
+            var waitForFile = Path.Combine(clientTentacle.TemporaryDirectory.DirectoryPath, "waitforme");
+
+            var startScriptCommand = new StartScriptCommandV2Builder()
+                .WithIsolation(ScriptIsolationLevel.FullIsolation)
+                .WithMutexName("bob")
+                .WithScriptBody(new ScriptBuilder()
+                    .Print("hello")
+                    .WaitForFileToExist(waitForFile)
+                    .Print("AllDone"))
+                .Build();
+
+            List<ProcessOutput> logs = new List<ProcessOutput>();
+            Assert.ThrowsAsync<HalibutClientException>(async () => await clientTentacle.TentacleClient.ExecuteScriptAssumingException(startScriptCommand, logs, CancellationToken));
+
+            // Let the script finish.
+            File.WriteAllText(waitForFile, "");
+
+            var allLogs = logs.JoinLogs();
+
+            allLogs.Should().NotContain("AllDone");
+            scriptServiceExceptions.StartScriptLatestException.Should().NotBeNull();
+            scriptServiceCallCounts.StartScriptCallCountStarted.Should().Be(1);
+            scriptServiceCallCounts.GetStatusCallCountStarted.Should().Be(0);
+            scriptServiceCallCounts.CompleteScriptCallCountStarted.Should().Be(0);
+
+            // We must ensure all script are complete, otherwise if we shutdown tentacle while running a script the build can hang.
```

Comment:
I don't think we need this as we never start the original script

Reply:
I'll defer to @LukeButters on this one, as this was a copy pasta from his SSv1 tests.

Reply:
As per Luke's other comment this needs to be left as the script has started.